whois-api/src/main/java/net/ripe/db/whois/api/fulltextsearch/ElasticFulltextSearch.java

```diff
@@ -167,8 +169,12 @@ private SearchResponse.Lst createHighlights(final SearchHit hit) {
         final SearchResponse.Lst documentLst = new SearchResponse.Lst(hit.getId());
         final List<SearchResponse.Arr> documentArrs = Lists.newArrayList();
 
-
-        hit.getHighlightFields().forEach((attribute, highlightField) -> {
+        for (Map.Entry<String, HighlightField> highlightFieldMap : hit.getHighlightFields().entrySet()) {
```

Comment:
Why drop the `forEach`? I think it's more readable and does the same thing as the for loop and variables

Reply:
I am using a foreach but in different way, in the other way you can not use the continue and I wanted to avoid and if/else that covers all of this logic.

Reply:
I prefer my approach that this one:


        hit.getHighlightFields().forEach((attribute, highlightField) -> {
            if (!("lookup-key".equals(attribute) || "lookup-key.custom".equals(attribute))){
                if(attribute.contains(".custom")) {
                    final SearchResponse.Arr arr = new SearchResponse.Arr(StringUtils.substringBefore(highlightField.name(), ".custom"));
                    arr.setStr(new SearchResponse.Str(null, StringUtils.join(highlightField.getFragments(), ",")));
                    documentArrs.add(arr);

                    //Somehow if searched term contains "." highlight field custom has no vlue for it.
                } else if(!hit.getHighlightFields().containsKey(attribute + ".custom"))  {
                    final SearchResponse.Arr arr = new SearchResponse.Arr(highlightField.name());
                    arr.setStr(new SearchResponse.Str(null, StringUtils.join(highlightField.getFragments(), ",")));
                    documentArrs.add(arr);
                }
            }
        });

Reply:
Use `return` rather than `continue` in a `forEach` loop

Reply:
yes, this should work. Ok, thank you! I will update